### PR TITLE
Fix parsing esds

### PR DESCRIPTION
**Pull request by Noarkhh**
- Fix parsing AudioSpecificConfig section (section 5) of esds atom
- Refactor duplicate code of parsing and generating esds and AudioSpecificConfig

### CODE REVIEW

**Review by mat-hek — APPROVED** [review]
LGTM

File: lib/membrane/aac/parser/config.ex [hunk 1]
```diff
@@ -23,10 +24,24 @@ defmodule Membrane.AAC.Parser.Config do
...
-        Map.merge(stream_format, Esds.parse_esds(esds))
+        esds_stream_format = Esds.parse_esds(esds)
 
-      {:audio_specific_config, config} ->
-        AudioSpecificConfig.parse_audio_specific_config(config)
+        if esds_stream_format.sample_rate != stream_format.sample_rate,
+          do:
+            Membrane.Logger.warning(
+              "Sample rate field decoded from esds differs from stream_format:\nesds:#{inspect(esds_stream_format)}\nstream_format:#{inspect(stream_format)}"
+            )
```
Comment (bblaszkow06) [inline lib/membrane/aac/parser/config.ex]: You can use multiline string here instead:
```suggestion
            Membrane.Logger.warning("""
            Sample rate field decoded from esds differs from stream_format:
            esds: #{inspect(esds_stream_format)}
            stream_format: #{inspect(stream_format)}
            """)

```

File: lib/membrane/aac/parser/config.ex [hunk 2]
```diff
@@ -23,10 +24,24 @@ defmodule Membrane.AAC.Parser.Config do
...
+
+        if esds_stream_format.channels != stream_format.channels,
+          do:
+            Membrane.Logger.warning(
+              "Channels field decoded from esds differs from stream_format:\nesds:#{inspect(esds_stream_format)}\nstream_format:#{inspect(stream_format)}"
+            )
```
Comment (bblaszkow06) [inline lib/membrane/aac/parser/config.ex]: A bit of nitpick, but these warnings are not very DRY and the duplicated log is overly verbose. Instead, you could iterate over fields and print a warning only once, e.g.:
```suggestion
        [:sample_rate, :channels]
        |> Enum.map(fn field ->
          {Map.fetch!(esds_stream_format, field),  Map.fetch!(stream_format, field)}
        end)
        |> Enum.each
          [] ->
            :ok

          fields ->
            Membrane.Logger.warning("""
            Fields #{inspect(fields)} decoded from esds differs from stream_format:
            esds: #{inspect(esds_stream_format)}
            stream_format: #{inspect(stream_format)}
            """)
        end
```
Alternatively, if printing separate warnings for each difference, I'd only print the different values, not whole structs, e.g.

```suggestion
        [:sample_rate, :channels]
        |> Enum.map(fn field ->
          {field, Map.fetch!(esds_stream_format, field), Map.fetch!(stream_format, field)}
        end)
        |> Enum.filter(fn {_field, esds_field, stream_field} -> esds_field != stream_field end)
        |> Enum.each(fn {field, esds_field, stream_field} ->
          Membrane.Logger.warning("""
          #{inspect(field)} decoded from esds differs from stream_format:
          esds: #{inspect(esds_field)}
          stream_format: #{inspect(stream_field)}
          """)
        end)

```

**Review by bblaszkow06 — APPROVED** [review]
Just a few notes on logging, leaving it up to you to decide how to handle them

**Pull request merged**